contrib/babelfishpg_tsql/src/tablecmds.c

```diff
@@ -97,42 +106,56 @@ void pre_check_trigger_schema(List *object, bool missing_ok){
 
 	tgscan = systable_beginscan(tgrel, TriggerRelidNameIndexId, false,
 									NULL, 1, &key);
-
 	while (HeapTupleIsValid(tuple = systable_getnext(tgscan)))
 	{
 		Form_pg_trigger pg_trigger = (Form_pg_trigger) GETSTRUCT(tuple);
-		
-		if (namestrcmp(&(pg_trigger->tgname), depname) == 0)
+		if(!OidIsValid(pg_trigger->tgrelid))
 		{
-			reloid = OidIsValid(pg_trigger->tgrelid) ? pg_trigger->tgrelid :
-						InvalidOid;
-			relation = RelationIdGetRelation(reloid); 
-			schema_oid = get_rel_namespace(reloid);
-            schema_name = get_namespace_name(schema_oid);
-			if ((list_length(object) > 1 && 
-			strcasecmp(schema_name , get_physical_schema_name(get_cur_db_name(),trigger_schema)) != 0 && !missing_ok)){
-				ereport(ERROR,
-				(errcode(ERRCODE_UNDEFINED_OBJECT),
-				errmsg("trigger \"%s.%s\" does not exist",
-						trigger_schema ,depname)));
-			} else if (list_length(object) == 1 && strcasecmp(schema_name, get_dbo_schema_name(get_cur_db_name())) != 0
-			&& !missing_ok){
-				ereport(ERROR,
-					(errcode(ERRCODE_UNDEFINED_OBJECT),
-					errmsg("trigger \"%s\" does not exist",
-						depname)));
+			reloid = InvalidOid;
+			break;
+		}
+		pg_trigger_physical_schema = get_namespace_name(get_rel_namespace(pg_trigger->tgrelid));
```

Comment:
This whole logic is same as the hook function, right? Can we make a separate function for this whole catalog lookup and use that in both places?